### ISSUE TITLE
Output format CSV bug

### ISSUE THREAD

**Issue by suriyaJaboon** [pteich/elastic-query-export#10]
### I think the position variable problem
 - package formats
 - file csv.go in line 56
```go
var document map[string]interface{}
```

### Field in hits _source
```json
{"@timestamp":"2020-12-31T23:03:49.000+07:00","acctstat":"Start","sessid":"01147A59A1332459"}
{"inoct":5990485,"outoct":90336295,"@timestamp":"2021-12-28T12:54:36.000+07:00","acctstat":"Stop","sessid":"011441190AC3E11D"}
{"inoct":15671390,"outoct":304085977,"@timestamp":"2021-12-30T13:30:59.000+07:00","acctstat":"Interim-Update","sessid":"0114411904F3F365"}
```

### Print data after json unmarshal
```go
map[@timestamp:2021-12-28T12:25:30.000+07:00 acctstat:Start inoct:3.4037472e+07 outoct:6.09485963e+08 sessid:011441190AC3E11D]
```

### Output .csv file **`result`**
| @timestamp                    | acctstat       | inoct    | outoct    | sessid           |
|-------------------------------|----------------|----------|-----------|------------------|
| 2021-12-28T20:16:10.000+07:00 | Interim-Update | 248744   | 39128     | 011441190CB3E9F2 |
| 2021-12-28T12:54:36.000+07:00 | Stop           | 5990485  | 90336295  | 011441190AC3E11D |
| 2021-12-28T12:25:30.000+07:00 | Start          | 25733507 | 358278252 | 011441190AC3E11D |

### Output .csv file **`expected`**
| @timestamp                    | acctstat       | inoct    | outoct    | sessid           |
|-------------------------------|----------------|----------|-----------|------------------|
| 2021-12-28T20:16:10.000+07:00 | Interim-Update | 248744   | 39128     | 011441190CB3E9F2 |
| 2021-12-28T12:54:36.000+07:00 | Stop           | 5990485  | 90336295  | 011441190AC3E11D |
| 2021-12-28T12:25:30.000+07:00 | Start          | | | 011441190AC3E11D |

**Comment by SteveyPugs** [comment]
Having the same issue here. CSV comes out all messed up.

**Comment by pteich** [comment]
Sorry @SteveyPugs @suriyaJaboon for the delay and apologies for the messed up output in CSV format if now fields are defined as input argument. I only needed to export selected fields and for this is worked and I had not the time to test the full document output.

This is now fixed. Therefore the field names of the document are collected and exported as header in the first line of the CSV file. All following lines now export the right field in the right place.

I'll create a new release after merging the PR.
